install/install.ps1

```diff
@@ -194,12 +206,10 @@ function Install-Post-0-4-0 {
                 $newPath = "${currentPath};${targetDirectory}"
             }
         }
-        # Update the $PATH environment variable with any modifications made above. We use `setx` here instead of
-        # directly writing back the registry value because `setx` signals new processes to pick up on the environment
-        # variable changes.
-        setx PATH $newPath > $null
+        Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+        Send-SettingChange

```

Comment:
Should we use `Set-ItemProperty` or `[Environment]::SetEnvironmentVariable`? Do you know if there's an important difference there, @Haard30?

Reply:
I found using the `Set-ItemProperty` well documented [here](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-itemproperty?view=powershell-7.2). I am not aware of any important difference but `[Environment]::SetEnvironmentVariable` directly uses .NET class `[Environment]` and its function.

Reply:
Those sound different, `[Enviornment]` using Dotnet's `Enviornment` under the hood implies only setting the env var for this process and not updating the Registry.

Reply:
I found this doc on `[Environment]` [http://xahlee.info/powershell/environment_variables.html](http://xahlee.info/powershell/environment_variables.html)